src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedLocalStorage.cs

```diff
@@ -15,6 +16,7 @@ namespace Microsoft.AspNetCore.Components.Web.Extensions
     ///
     /// See: https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage
     /// </summary>
+    [UnsupportedOSPlatform("browser")]
     public class ProtectedLocalStorage : ProtectedBrowserStorage
```

Comment:
if `ProtectedBrowserStorage` has `[UnsupportedOSPlatform("browser")]` applied and `ProtectedLocalStorage` derives from it, then most probably there is no need to apply it to `ProtectedLocalStorage` and other deriving classes (`ProtectedSessionStorage` in this case)

@buyaa-n am I right?

Reply:
The attributes are defined as [`Inherited = false`](https://github.com/dotnet/runtime/blob/06963eaedfe666c087a9fabb89b3f4bca57634c7/src/libraries/System.Private.CoreLib/src/System/Runtime/Versioning/PlatformAttributes.cs#L130). So i guess `ISymbol.GetAttrbiutes()` wouldn't return parent attributes and the analyzer not loading parent attributes manually. So guess we should attribute derived types

Reply:
@buyaa-n thanks for a great answer!